Subject: [PATCH] ExpPreg patch, one litter per father
---
Index: src/com/lilithsthrone/game/character/GameCharacter.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/com/lilithsthrone/game/character/GameCharacter.java b/src/com/lilithsthrone/game/character/GameCharacter.java
--- a/src/com/lilithsthrone/game/character/GameCharacter.java	(revision 3d5bc6ae36ec8db07ae6775fe951378ecce84c5f)
+++ b/src/com/lilithsthrone/game/character/GameCharacter.java	(date 1742066365841)
@@ -21243,27 +21243,7 @@
 					for (Litter litter : this.getPregnantLitters()) {
 						// Find litter father if there's one
 						if (litter.getFather()!=null && partner != null && litter.getFather() == partner) {
-							List<OffspringSeed> offspring = new ArrayList<>(litter.getOffspringSeed()); // Add children here: 
-							for (int i = 0; i < numberOfChildren; i++) { 
-								OffspringSeed os = new OffspringSeed(this, partner, partnerBody);
-								offspring.add(os);
-								try {
-									Main.game.addOffspringSeed(os, false);
-								} catch (Exception e) {
-									e.printStackTrace();
-								}
-							}
-							// Other initial litter data: 
-							fertilisationType = litter.getFertilisationType();
-							LocalDateTime conceptionDate = litter.getConceptionDate();
-							LocalDateTime birthDate = litter.getBirthDate();
-							
-							Litter newLitter = new Litter(conceptionDate, birthDate, this, partner, fertilisationType, offspring);
-							pregnantLitters.remove(litter);
-							pregnantLitters.add(newLitter);
-							pregnantLitter = pregnantLitters.get(0); // In case it's the first litter
-							checkNewLitter = false;
-							break;
+                            return pregnancyDescription;
 						}
 					}
 					
