Subject: [PATCH] Decrease experience from orgasmic level drain
---
Index: src/com/lilithsthrone/game/character/GameCharacter.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/com/lilithsthrone/game/character/GameCharacter.java b/src/com/lilithsthrone/game/character/GameCharacter.java
--- a/src/com/lilithsthrone/game/character/GameCharacter.java	(revision 0ecb800b50023ddce1383f1b0e9b18b8e518932f)
+++ b/src/com/lilithsthrone/game/character/GameCharacter.java	(date 1742062949365)
@@ -10823,7 +10823,7 @@
 						+ "<br/>[style.italicsBad(As [npc.she] [npc.verb(orgasm)], [npc.name] [npc.verb(feel)] [npc.herself] getting weaker...)]"
 					+ "</p>"
 					+ target.levelDown(1)
-					+ this.incrementExperience(exp, false));
+					+ this.incrementExperience(exp/4, false));
 			
 		} else {
 			return UtilText.parse(target, this,
